pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -619,6 +619,19 @@ func (m *kubeGenericRuntimeManager) killContainer(pod *v1.Pod, containerID kubec
 		gracePeriod = *pod.DeletionGracePeriodSeconds
 	case pod.Spec.TerminationGracePeriodSeconds != nil:
 		gracePeriod = *pod.Spec.TerminationGracePeriodSeconds
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.ProbeTerminationGracePeriod) {
```

Comment:
can we add unit testing for this scenario in a follow-on, didnt see it in TestKillContainer

Reply:
I think we should evaluate handling this change similar to DownwardAPI support for hugepages.
https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2053-downward-api-hugepages/kep.yaml#L41

kubelet did not gate behavior on the feature gate, just the persistence of the field on the api server was controlled by the gate.

Reply:
I think we could do the following:

when this feature moves to beta, its off by default, and the feature gate is no longer used in the kubelet.
2 releases later the feature is on by default.